fingertip/plugins/script.py

```diff
@@ -1,24 +1,28 @@
 # Licensed under GNU General Public License v3 or later, see COPYING.
 # Copyright (c) 2019 Red Hat, Inc., see CONTRIBUTORS.
 
+import sys
 import fingertip
 
 
-# fingertip ... + script.run script  -  no cache, never persist, always rerun
-# fingertip ... + script.run script + ssh  -  cache just for this invocation
-# fingertip ... + script.run script --cache=1h  -  try to cache for 1h at most
-# fingertip ... + script.run script --cache=1h + ssh  -  try reusing the cache
-# fingertip ... + transient script.run script --cache=1h + ssh  -  revert
-
-
-def _should_run_be_transient(scriptpath, cache=0, no_unseal=False):
+# script.X (X in [run, setup]) without --cache=Y defaults to:
+#  * not caching if it is the last plugin in the pipeline
+#  * caching if there are further plugins in the pipeline
+# Examples:
+# fingertip ... + script.X script  -  no cache, never persist, always rerun
+# fingertip ... + script.X script + ssh  -  cache just for this invocation
+# fingertip ... + script.X script --cache=1h  -  try to cache for 1h at most
+# fingertip ... + script.X script --cache=1h + ssh  -  try reusing the cache
+# fingertip ... + transient script.X script --cache=1h + ssh  -  revert
+def _should_be_transient(scriptpath, cache=0, no_unseal=False):
     return False if cache else 'last'
 
 
-@fingertip.transient(when=_should_run_be_transient)
-def run(m, scriptpath, cache=0, no_unseal=False):
-    m.log.debug(f'Plugin: script, scriptpath: {scriptpath}, '
-                f'cache: {cache}, no_unseal: {no_unseal}')
+def _exec(m, scriptpath, mode=None, last_step=None, cache=0, no_unseal=False):
+    m.log.debug(f'Plugin: script, mode: {mode}, scriptpath: {scriptpath}, '
```

Comment:
outdated plugin name

Reply:
I don't understand. What is the current name?

Reply:
It depends on... ah =) Well, I call whatever you are supposed to write after `fingertip` plugins (so, `script.run` and `script.test`, not `script`). Never mind.